File: 프로그래머스/0/120841. 점의 위치 구하기/점의 위치 구하기.py
def solution(dot):
    answer = 0  # 사분면 값을 저장할 변수 초기화

    if dot[0] > 0 and dot[1] > 0:  # x > 0, y > 0 → 1사분면
        answer = 1
    elif dot[0] < 0 and dot[1] > 0:  # x < 0, y > 0 → 2사분면
        answer = 2
    elif dot[0] < 0 and dot[1] < 0:  # x < 0, y < 0 → 3사분면
        answer = 3
    elif dot[0] > 0 and dot[1] < 0:  # x > 0, y < 0 → 4사분면
        answer = 4

    return answer  # 판별된 사분면 반환

# 테스트 예제 실행
print(solution([2, 4]))  # 1
print(solution([-7, 9])) # 2

File: 프로그래머스/0/120841. 점의 위치 구하기/점의 위치 구하기.go
// 사분면 판별 함수
func solution(dot []int) int {
    if dot[0] > 0 && dot[1] > 0 {
        return 1
    } else if dot[0] < 0 && dot[1] > 0 {
        return 2
    } else if dot[0] < 0 && dot[1] < 0 {
        return 3
    } else if dot[0] > 0 && dot[1] < 0 {
        return 4
    }
    return 0 // 입력이 0인 경우 등 예외 처리
}